syntax = "proto3";
option optimize_for = LITE_RUNTIME;
option go_package = "github.com/GameWorkstore/async-network-engine-go;asyncnetworkengine";

package GameWorkstore.AsyncNetworkEngine;

enum Transmission
{
   NotSpecified = 0;
   ErrorConnection = 1;
   ErrorProtocol = 2;
   ErrorParser = 3;

   Success = 200; // StatusOK
   ErrorNoData = 204; //StatusNoContent
   ErrorDecode = 400; //StatusBadRequest
   ErrorMethodNotAllowed = 405; //StatusMethodNotAllowed
   ErrorEncode = 424; //StatusFailedDependency
   ErrorInternalServer = 500; //StatusInternalServerError
   ErrorNotImplemented = 501; //StatusNotImplemented
   ErrorRequestTimeout = 408; //StatusRequestTimeout
}

message GenericRequest
{
   string Messege = 1;
}

message GenericResponse
{
   string Messege = 2;
}

message GenericErrorResponse
{
   string Error = 1;
}
